src/CMakeLists.txt

```diff
@@ -79,12 +81,12 @@ if(CMAKE_INTERPROCEDURAL_OPTIMIZATION)
     # Policy that enables INTERPROCEDURAL_OPTIMIZATION
     set(CMAKE_POLICY_DEFAULT_CMP0069 NEW)
     cmake_policy(SET CMP0069 NEW)
-    set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
     if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
       # Clang has some issues with thinLTO (segfaults on final linking)
       # This enables full LTO through CMake
       # FullLTO is slow, but may yield performance improvements.
-      add_compile_options(-flto=full)
+      set(CMAKE_CXX_COMPILE_OPTIONS_IPO "-flto")
```

Comment:
Did you investigate parallel LTO? `-flto=8` for example. GCC has it, not sure if Clang as well. It makes everything faster but also eats a lot more RAM that way, so may cause issues on memory-constrained systems.

(This is exactly why I was hesitating to enable LTO everywhere, because it's either annoyingly slow or memory-hungry. A solution would be to ask the buildsystem to know how many parallelization slots are available (`-flto=jobserver`), which works with Make, but not with Ninja yet.)

Reply:
@mosra It's not enabled by default in CMake, but I went ahead and enabled it.

Reply:
Ah yeah, it breaks Clang-Tidy which is annoying: https://app.circleci.com/pipelines/github/facebookresearch/habitat-sim/6367/workflows/5d3ff9c9-d972-4913-ba17-17f8e91d68ee/jobs/30212